math/wide_integer/uintwide_t.h

```diff
@@ -1863,45 +1906,45 @@
     static constexpr size_t wr_string_max_buffer_size_dec = (20U + size_t((std::uintmax_t(my_width2) * UINTMAX_C(301)) / UINTMAX_C(1000))) + 1U;
 
     // Write string function.
-    WIDE_INTEGER_CONSTEXPR bool wr_string(      char*              str_result,
+    WIDE_INTEGER_CONSTEXPR auto wr_string(      char*              str_result, // NOLINT(readability-function-cognitive-complexity)
                                           const std::uint_fast8_t  base_rep     = 0x10U,
                                           const bool               show_base    = true,
                                           const bool               show_pos     = false,
                                           const bool               is_uppercase = true,
                                                 unsinged_fast_type field_width  = 0U,
-                                          const char               fill_char    = char('0')) const
+                                          const char               fill_char    = char('0')) const -> bool
     {
       bool wr_string_is_ok = true;
 
-      if(base_rep == 8U)
+      if(base_rep == UINT8_C(8))
       {
         uintwide_t t(*this);
 
-        const limb_type mask(std::uint8_t(0x7U));
+        const auto mask = static_cast<limb_type>(std::uint8_t(0x7U));
 
-        char str_temp[std::size_t(wr_string_max_buffer_size_oct)];
+        std::array<char, std::size_t(wr_string_max_buffer_size_oct)> str_temp { };
 
         unsinged_fast_type pos = (sizeof(str_temp) - 1U);
 
         if(t.is_zero())
         {
           --pos;
 
-          str_temp[pos] = char('0');
+          str_temp.at(static_cast<std::size_t>(pos)) = char('0');
```

Comment:
Note that the `at` APIs typically throw on out-of-bounds indices. Unless you intend to pass in out-of-bounds indices during correct execution of the code (and then to catch the exceptions), then `at` is maybe the wrong API to use.

I would consider silencing the warning instead. (I think the proscribed solution is to use the GSL library, which you may be reluctant to do!)

Reply:
Yes. I was concerned about these. I feel that, based on your review, i might revert the uses of `at()`. There were a bunch of such uses. But `at()` can throw while range-checking so i think that's not good either.

I'm actually going to revert and locally disable a bunch of these based on this review point.

Reply:
Done. For array access, switch back to random access `operator[](size_t)`. Disable messages via `NOLINT(...)` where needed.